### ISSUE TITLE
PLR1708 false positives on nested functions

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#21162]
### Summary

[`stop-iteration-return` (PLR1708)](https://docs.astral.sh/ruff/rules/stop-iteration-return/) reports false positives when a non-generator function raises `StopIteration` and contains a nested generator function, or a generator function contains a nested non-generator function that raises `StopIteration`. [Example](https://play.ruff.rs/2218cd6c-d01f-422b-a5a3-469d1b9c0cbf):
```console
$ cat >plr1708.py <<'# EOF'
def f():
    def g():
        yield 1
    raise StopIteration

def g():
    def f():
        raise StopIteration
    yield 1
# EOF

$ ruff --isolated check plr1708.py --select PLR1708 --preview --output-format concise -q
plr1708.py:4:5: PLR1708 Explicit `raise StopIteration` in generator
plr1708.py:8:9: PLR1708 Explicit `raise StopIteration` in generator
```

### Version

ruff 0.14.3 (8737a2d5f 2025-10-30)
